File: Frontend/graphql/query/document.ts
import { gql } from '@apollo/client';

export const GET_MY_DOCUMENTS = gql`
  query GetMyDocuments($employeeId: String!) {
    getMyDocuments(employeeId: $employeeId) {
      id
      employeeId
      employeeName
      category
      name
      url
      expiryDate
      status
      rejectionReason
      createdAt
      updatedAt
    }
  }
`;

// Made with Bob

File: Backend/Modules/DocumentFeature/DocumentFeature.Domain/Document.cs
using HRMS.Core.Postgres.Common;
using DocumentFeature.Domain.Enums;

namespace DocumentFeature.Domain
{
    public class Document : BaseEntity
    {
        public string EmployeeId { get; set; } = string.Empty;
        public DocumentCategory Category { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Url { get; set; }
        public DateTime? ExpiryDate { get; set; }
        public DocumentStatus Status { get; set; } = DocumentStatus.Missing;
        public string? RejectionReason { get; set; }
    }
}

// Made with Bob
